src/dune_engine/sandbox_mode.ml

```diff
@@ -69,42 +76,39 @@ end
 module Set = struct
   type key = t
 
-  type t = bool Dict.t
+  type t = int
 
-  let compare = Dict.compare Bool.compare
+  let compare = Int.compare
 
-  let of_func = Dict.of_func
+  let singleton k = 1 lsl to_int k
```

Comment:
I wonder if we could factor out the bit-twiddling from this module into a new Stdune module, say `Bit_dict`. Otherwise, this refactoring makes this module pretty hard to understand.

Reply:
Ah yes, will give that a try.